tools/ci/ci-nightly.yaml

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root for license information.
+
+# MixedReality-SpectatorView build native content
+
+# Scheduled run, always run to monitor churn in external dependencies
+schedules:
+- cron: "0 0 * * *"
+  displayName: Daily Midnight Build
+  branches:
+    include:
+    - master
+  always: true
+
+# Trigger CI on push changes
+trigger: none
+
+# Do not trigger CI on PRs
+pr: none
+
+# Give a unique name to the build each time it runs
+name: ci-nightly-$(SourceBranchName)-$(Date:yyyyMMdd)-$(Rev:r)
+
+variables:
```

Comment:
Nit: In general stick variables at the top so that you can see the key bits of data going into each YAML file (i.e. almost like function signature)

https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#set-variables-in-pipeline

In this case, I guess it's LESS interesting because UnityVersion is more or less a const